server/src/main/java/org/elasticsearch/index/engine/Segment.java

```diff
@@ -170,6 +186,12 @@ public void writeTo(StreamOutput out) throws IOException {
         if (hasAttributes) {
             out.writeMap(attributes, StreamOutput::writeString);
         }
+        out.writeString(codec);
+        boolean hasKnnFormats = knnFormats != null;
+        out.writeBoolean(hasKnnFormats);
+        if (hasKnnFormats) {
+            out.writeMap(knnFormats, StreamOutput::writeString);
+        }
```

Comment:
All of these need to be bwc protected for transport serialization. I think if we update `attrbutes` we should be OK?

Reply:
yeah, one more reason for putting them into `attributes`, thanks ;)